### PR TITLE
352 aci leaf access bundle policy group

### LINKED ISSUE #352: aci_leaf_access_bundle_policy_group and override policy groups
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Module aci_leaf_access_bundle_policy_group is currently missing attribute for attaching override policy group.

https://pubhub.devnetcloud.com/media/apic-mim-ref-301/docs/MO-infraAccBndlSubgrp.html

`dn=uni/infra/funcprof/accbundle-xxxx/accsubbndl-xxxx`

### PR DESCRIPTION
**Pull request by akinross**
fixes #352

### CODE REVIEW

File: aci/resource_aci_lacpifpol.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendAttrSchemas(map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"prio": {
```
Comment (sajagana) [inline aci/resource_aci_lacpifpol.go:33]:
```suggestion
			"priority": {
```
Update all other references.

File: aci/resource_aci_lacpifpol.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"tx_rate": {
```
Comment (sajagana) [inline aci/resource_aci_lacpifpol.go:38]:
```suggestion
			"transmit_rate": {
```
Update all other references.

File: website/docs/d/lacp_member_policy.markdown [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Argument Reference ##
+
+* `name` - (Required) Name of object LACP Member Policy.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the LACP Member Policy.
+* `annotation` - (Optional) Annotation of object LACP Member Policy.
+* `name_alias` - (Optional) Name Alias of object LACP Member Policy.
+* `name_alias` - (Optional) Name alias.
```
Comment (sajagana) [inline website/docs/d/lacp_member_policy.markdown:39]:
```suggestion
* `annotation` - (Optional) Annotation of the object LACP Member Policy.
* `name_alias` - (Optional) Name Alias of the object LACP Member Policy.
```

File: website/docs/d/lacp_member_policy.markdown [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+* `name` - (Required) Name of object LACP Member Policy.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the LACP Member Policy.
+* `annotation` - (Optional) Annotation of object LACP Member Policy.
+* `name_alias` - (Optional) Name Alias of object LACP Member Policy.
+* `name_alias` - (Optional) Name alias. 
+* `prio` - (Optional) Priority. Port priority - LACP uses the port priority to decide which ports should be put in standby mode when there is a limitation that prevents all compatible ports from aggregating and which ports should be put into active mode. A higher port priority value means a lower priority for LACP
+* `tx_rate` - (Optional) Transmission Rate. The configured transmit rate of the LACP packets.
```
Comment (sajagana) [inline website/docs/d/lacp_member_policy.markdown:41]:
```suggestion
* `priority` - (Optional) Priority. Port priority - LACP uses the port priority to decide which ports should be put in standby mode when there is a limitation that prevents all compatible ports from aggregating and which ports should be put into active mode. A higher port priority value means a lower priority for LACP
* `transmit_rate` - (Optional) Transmission Rate. The configured transmit rate of the LACP packets.
```

File: website/docs/d/leaf_access_bundle_policy_sub_group.markdown [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## Argument Reference ##
+
+* `leaf_access_bundle_policy_group_dn` - (Required) Distinguished name of parent infraAccBndlGrp object.
+* `name` - (Required) Name of object Override Policy Group.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Override Policy Group.
+* `annotation` - (Optional) Annotation of object Override Policy Group.
+* `name_alias` - (Optional) Name Alias of object Override Policy Group.
```
Comment (sajagana) [inline website/docs/d/leaf_access_bundle_policy_sub_group.markdown:40]:
```suggestion
* `leaf_access_bundle_policy_group_dn` - (Required) Distinguished name of the parent infraAccBndlGrp object.
* `name` - (Required) Name of the object Override Policy Group.

## Attribute Reference ##
* `id` - Attribute id set to the Dn of the Override Policy Group.
* `annotation` - (Optional) Annotation of the object Override Policy Group.
* `name_alias` - (Optional) Name Alias of the object Override Policy Group.
```

File: website/docs/r/lacp_member_policy.markdown [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+* `Location` - Fabric - Access Policies - Policies - Interface - Port Channel Member
+
+## Example Usage ##
+
+```hcl
+resource "aci_lacp_member_policy" "example" {
+  name  = "example"
+  prio = "32768"
+  tx_rate = "normal"
```
Comment (sajagana) [inline website/docs/r/lacp_member_policy.markdown:29]:
```suggestion
  priority = "32768"
  transmit_rate = "normal"
```

File: website/docs/r/lacp_member_policy.markdown [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+## Argument Reference ##
+
+* `name` - (Required) Name of the object LACP Member Policy.
+* `annotation` - (Optional) Annotation of the object LACP Member Policy.
+* `description` - (Optional) Description of the object LACP Member Policy.
+* `name_alias` - (Optional) Name alias.
+* `prio` - (Optional) Priority.Port priority - LACP uses the port priority to decide which ports should be put in standby mode when there is a limitation that prevents all compatible ports from aggregating and which ports should be put into active mode. A higher port priority value means a lower priority for LACP Allowed range is 1-65535 and default value is "32768".
+* `tx_rate` - (Optional) Transmission Rate.The configured transmit rate of the LACP packets. Allowed values are "fast", "normal", and default value is "normal". Type: String.
```
Comment (sajagana) [inline website/docs/r/lacp_member_policy.markdown:40]:
```suggestion
* `priority` - (Optional) Priority.Port priority - LACP uses the port priority to decide which ports should be put in standby mode when there is a limitation that prevents all compatible ports from aggregating and which ports should be put into active mode. A higher port priority value means a lower priority for LACP Allowed range is 1-65535 and default value is "32768".
* `transmit_rate` - (Optional) Transmission Rate.The configured transmit rate of the LACP packets. Allowed values are "fast", "normal", and default value is "normal". Type: String.
```

**Review by sajagana — CHANGES_REQUESTED** [review]

File: website/docs/d/lacp_member_policy.markdown [hunk 8]
```diff
@@ -0,0 +1,41 @@
+---
+subcategory: "Access Policies"
+layout: "aci"
+page_title: "ACI: aci_lacpmember_policy"
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:4]: ok

File: website/docs/d/lacp_member_policy.markdown [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+* `Distinguished Name` - uni/infra/lacpifp-{name}
+
+## GUI Information ##
+
+* `Location` - Fabric - Access Policies - Policies - Interface - Port Channel Member
+
+## Example Usage ##
+
+```hcl
+data "aci_lacpmember_policy" "example" {
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:26]: ok

File: website/docs/d/lacp_member_policy.markdown (hunk 3 above)
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:39]: ok

File: website/docs/d/lacp_member_policy.markdown (hunk 4 above)
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:41]: ok

File: website/docs/d/leaf_access_bundle_policy_sub_group.markdown (hunk 5 above)
Comment (akinross) [reply website/docs/d/leaf_access_bundle_policy_sub_group.markdown:40]: ok

File: website/docs/r/lacp_member_policy.markdown (hunk 6 above)
Comment (akinross) [reply website/docs/r/lacp_member_policy.markdown:29]: ok

File: website/docs/r/lacp_member_policy.markdown [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+## Importing ##
+
+An existing LACPMemberPolicy can be [imported][docs-import] into this resource via its Dn, via the following command:
+[docs-import]: https://www.terraform.io/docs/import/index.html
+
+```
+terraform import aci_lacpmember_policy.example <Dn>
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/r/lacp_member_policy.markdown:48]: ok

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+* `Location` - Fabric - Access Policies - Interfaces - Leaf Interfaces - Policy Groups - [ PC | VPC ] Interface - Advanced Policies - Override Access Policy Groups
+
+## Example Usage ##
+
+```hcl
+resource "aci_leaf_access_bundle_policy_sub_group" "example" {
+  leaf_access_bundle_policy_group_dn  = aci_leaf_access_bundle_policy_group.example.id
+  name  = "example"
+  port_channel_member = aci_resource.example.id
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/r/leaf_access_bundle_policy_sub_group.markdown:29]: ok

File: aci/data_source_aci_infraaccbndlsubgrp.go [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+		SchemaVersion: 1,
+		Schema: AppendAttrSchemas(map[string]*schema.Schema{
+			"leaf_access_bundle_policy_group_dn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/data_source_aci_infraaccbndlsubgrp.go:25]: ok

File: aci/data_source_aci_lacpifpol.go [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+func dataSourceAciLACPMemberPolicy() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceAciLACPMemberPolicyRead,
+		SchemaVersion: 1,
+		Schema: AppendAttrSchemas(map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"prio": {
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/data_source_aci_lacpifpol.go:22]: ok

File: aci/data_source_aci_lacpifpol.go [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"tx_rate": {
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/data_source_aci_lacpifpol.go:27]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+
+	"github.com/ciscoecosystem/aci-go-client/client"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAciOverridePolicyGroup_Basic(t *testing.T) {
+	var override_policy_group models.OverridePolicyGroup
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:15]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+					testAccCheckAciOverridePolicyGroupExists("aci_leaf_access_bundle_policy_sub_group.test_policy_sub_group", &override_policy_group),
+					testAccCheckAciOverridePolicyGroupAttributes(infra_acc_bndl_grp_name, infra_acc_bndl_subgrp_name, description, &override_policy_group),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAciOverridePolicyGroup_Update(t *testing.T) {
+	var override_policy_group models.OverridePolicyGroup
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:37]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+func testAccCheckAciOverridePolicyGroupConfig_basic(infra_acc_bndl_grp_name, infra_acc_bndl_subgrp_name string) string {
+	return fmt.Sprintf(`
+
+	resource "aci_leaf_access_bundle_policy_group" "test_policy_group" {
+		name 		= "%s"
+		description = "aci_leaf_access_bundle_policy_group created while acceptance testing"
+
+	}
+
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:73]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+	resource "aci_leaf_access_bundle_policy_sub_group" "test_policy_sub_group" {
+		name 		= "%s"
+		description = "override_policy_group created while acceptance testing"
+		port_channel_member = aci_leaf_access_bundle_policy_group.test_policy_group.id
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:77]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	`, infra_acc_bndl_grp_name, infra_acc_bndl_subgrp_name)
+}
+
+func testAccCheckAciOverridePolicyGroupExists(name string, override_policy_group *models.OverridePolicyGroup) resource.TestCheckFunc {
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:83]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+		}
+
+		client := testAccProvider.Meta().(*client.Client)
+
+		cont, err := client.Get(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		override_policy_groupFound := models.OverridePolicyGroupFromContainer(cont)
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:102]: ok

File: aci/resource_aci_infraaccbndlsubgrp_test.go [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+		return nil
+	}
+}
+
+func testAccCheckAciOverridePolicyGroupDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*client.Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type == "aci_leaf_access_bundle_policy_sub_group" {
+			cont, err := client.Get(rs.Primary.ID)
+			override_policy_group := models.OverridePolicyGroupFromContainer(cont)
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp_test.go:116]: ok

File: aci/resource_aci_lacpifpol.go (hunk 1 above)
Comment (akinross) [reply aci/resource_aci_lacpifpol.go:33]: ok

File: aci/resource_aci_lacpifpol.go (hunk 2 above)
Comment (akinross) [reply aci/resource_aci_lacpifpol.go:38]: ok

File: aci/resource_aci_lacpifpol.go [hunk 22]
```diff
@@ -0,0 +1,209 @@
...
+
+	if Prio, ok := d.GetOk("prio"); ok {
+		lacpIfPolAttr.Prio = Prio.(string)
+	}
+
+	if TxRate, ok := d.GetOk("tx_rate"); ok {
+		lacpIfPolAttr.TxRate = TxRate.(string)
+	}
+	lacpIfPol := models.NewLACPMemberPolicy(fmt.Sprintf(models.RnlacpIfPol, name), models.ParentDnlacpIfPol, desc, lacpIfPolAttr)
+	lacpIfPol.Status = "modified"
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_lacpifpol.go:163]: ok

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 23]
```diff
@@ -0,0 +1,255 @@
...
+	if Name, ok := d.GetOk("name"); ok {
+		infraAccBndlSubgrpAttr.Name = Name.(string)
+	}
+
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		infraAccBndlSubgrpAttr.NameAlias = NameAlias.(string)
+	}
+
+	infraAccBndlSubgrp := models.NewOverridePCVPCPolicyGroup(fmt.Sprintf(models.RninfraAccBndlSubgrp, name), LeafAccessBundlePolicyGroupDn, desc, infraAccBndlSubgrpAttr)
+	infraAccBndlSubgrp.Status = "modified"
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:171]: ok

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 24]
```diff
@@ -0,0 +1,255 @@
...
+		return diag.FromErr(err)
+	}
+	checkDns := make([]string, 0, 1)
+
+	if relationToinfraRsLacpInterfacePol, ok := d.GetOk("port_channel_member"); ok {
+		relationParam := relationToinfraRsLacpInterfacePol.(string)
+		checkDns = append(checkDns, relationParam)
+	}
+
+	d.Partial(true)
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:128]: not sure this was from auto-gen code, have removed and tested now

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 25]
```diff
@@ -0,0 +1,255 @@
...
+	}
+
+	checkDns := make([]string, 0, 1)
+
+	if d.HasChange("port_channel_member") || d.HasChange("annotation") {
+		_, newRelParam := d.GetChange("port_channel_member")
+		checkDns = append(checkDns, newRelParam.(string))
+	}
+
+	d.Partial(true)
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:185]: not sure this was from auto-gen code, have removed and tested now

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 26]
```diff
@@ -0,0 +1,255 @@
...
+		return nil, err
+	}
+	infraAccBndlSubgrp := models.OverridePCVPCPolicyGroupFromContainer(infraAccBndlSubgrpCont)
+	if infraAccBndlSubgrp.DistinguishedName == "" {
+		return nil, fmt.Errorf("OverridePCVPCPolicyGroup %s not found", dn)
+	}
+	return infraAccBndlSubgrp, nil
+}
+
+func setOverridePCVPCPolicyGroupAttributes(infraAccBndlSubgrp *models.OverridePCVPCPolicyGroup, d *schema.ResourceData) (*schema.ResourceData, error) {
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:58]: fix and fixed other importing issue regarding annotation of object

File: website/docs/r/lacp_member_policy.markdown [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+  priority = "32768"
+  transmit_rate = "normal"
+}
+```
+
+## Argument Reference ##
+
+* `id` - Attribute id set to the Dn of the LACP Member Policy.
+* `annotation` - (Optional) Annotation of the object LACP Member Policy.
+* `name_alias` - (Optional) Name Alias of the object LACP Member Policy.
```
Comment (anvitha-jain) [inline website/docs/r/lacp_member_policy.markdown:37]: Same as before

File: website/docs/r/lacp_member_policy.markdown [hunk 28]
```diff
@@ -0,0 +1,48 @@
...
+* `priority` - (Optional) Priority. Port priority - LACP uses the port priority to decide which ports should be put in standby mode when there is a limitation that prevents all compatible ports from aggregating and which ports should be put into active mode. A higher port priority value means a lower priority for LACP
```
Comment (anvitha-jain) [inline website/docs/r/lacp_member_policy.markdown:38]: Same as before

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+  name  = "example"
+  port_channel_member = aci_lacp_member_policy.example.id
+}
+```
+
+## Argument Reference ##
+
+* `leaf_access_bundle_policy_group_dn` - (Required) Distinguished name of the parent infraAccBndlGrp object.
+* `name` - (Required) Name of the object Override Policy Group.
+* `annotation` - (Optional) Annotation of the object Override Policy Group.
```
Comment (anvitha-jain) [inline website/docs/r/leaf_access_bundle_policy_sub_group.markdown:37]: Same as before

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown [hunk 30]
```diff
@@ -0,0 +1,49 @@
...
+* `description` - (Optional) Description of the object Override Policy Group.
```
Comment (anvitha-jain) [inline website/docs/r/leaf_access_bundle_policy_sub_group.markdown:38]: Same as before

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown [hunk 31]
```diff
@@ -0,0 +1,49 @@
...
+* `name_alias` - (Optional) Name alias.
```
Comment (anvitha-jain) [inline website/docs/r/leaf_access_bundle_policy_sub_group.markdown:39]: Change -- Name alias. -> Name alias of the Override Policy Group object.

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

File: go.mod [hunk 32]
```diff
@@ -3,7 +3,7 @@ module github.com/terraform-providers/terraform-provider-aci
 go 1.12
 
 require (
-	github.com/ciscoecosystem/aci-go-client v1.45.3
+	github.com/ciscoecosystem/aci-go-client v1.49.0
```
[the remark this answers is not in the document]
Comment (akinross) [reply go.mod:6]: ok

File: website/docs/d/lacp_member_policy.markdown [hunk 33]
```diff
@@ -0,0 +1,40 @@
...
+}
+```
+
+## Argument Reference ##
+
+* `name` - (Required) Name of object LACP Member Policy.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the LACP Member Policy.
+* `annotation` - (Optional) Annotation of the object LACP Member Policy.
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:37]: ok

File: website/docs/d/lacp_member_policy.markdown [hunk 34]
```diff
@@ -0,0 +1,40 @@
...
+* `name_alias` - (Optional) Name Alias of the object LACP Member Policy.
+* `priority` - (Optional) Priority. Port priority - LACP uses the port priority to decide which ports should be put in standby mode when there is a limitation that prevents all compatible ports from aggregating and which ports should be put into active mode. A higher port priority value means a lower priority for LACP
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:39]: ok

File: website/docs/d/lacp_member_policy.markdown [hunk 35]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+## Argument Reference ##
+
+* `name` - (Required) Name of object LACP Member Policy.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the LACP Member Policy.
+* `annotation` - (Optional) Annotation of the object LACP Member Policy.
+* `name_alias` - (Optional) Name Alias of the object LACP Member Policy.
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:38]: ok

File: website/docs/d/leaf_access_bundle_policy_sub_group.markdown [hunk 36]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+## Argument Reference ##
+
+* `leaf_access_bundle_policy_group_dn` - (Required) Distinguished name of the parent infraAccBndlGrp object.
+* `name` - (Required) Name of the object Override Policy Group.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Override Policy Group.
+* `annotation` - (Optional) Annotation of the object Override Policy Group.
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/leaf_access_bundle_policy_sub_group.markdown:39]: ok

File: website/docs/r/lacp_member_policy.markdown [hunk 37]
```diff
@@ -0,0 +1,48 @@
...
+  name  = "example"
+  priority = "32768"
+  transmit_rate = "normal"
+}
+```
+
+## Argument Reference ##
+
+* `id` - Attribute id set to the Dn of the LACP Member Policy.
+* `annotation` - (Optional) Annotation of the object LACP Member Policy.
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/r/lacp_member_policy.markdown:36]: ok

File: website/docs/d/leaf_access_bundle_policy_sub_group.markdown [hunk 38]
```diff
@@ -0,0 +1,40 @@
...
+* `name_alias` - (Optional) Name Alias of the object Override Policy Group.
```
[the remark this answers is not in the document]
Comment (akinross) [reply website/docs/d/leaf_access_bundle_policy_sub_group.markdown:40]: ok

File: website/docs/r/lacp_member_policy.markdown (hunk 28 above)
Comment (akinross) [reply website/docs/r/lacp_member_policy.markdown:38]: ok

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown (hunk 29 above)
Comment (akinross) [reply website/docs/r/leaf_access_bundle_policy_sub_group.markdown:37]: ok

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown (hunk 30 above)
Comment (akinross) [reply website/docs/r/leaf_access_bundle_policy_sub_group.markdown:38]: ok

File: website/docs/r/leaf_access_bundle_policy_sub_group.markdown (hunk 31 above)
Comment (akinross) [reply website/docs/r/leaf_access_bundle_policy_sub_group.markdown:39]: ok

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 39]
```diff
@@ -0,0 +1,279 @@
...
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"port_channel_member": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:39]: Why would that be the case?

**Review by anvitha-jain — DISMISSED** [review]
LGTM

**Review by sajagana — COMMENTED** [review]
LGTM!

File: website/docs/d/lacp_member_policy.markdown [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+
+```hcl
+data "aci_lacp_member_policy" "example" {
+  name  = "example"
+}
+```
+
+## Argument Reference ##
+
+* `name` - (Required) Name of LACP Member Policy object.
```
Comment (lhercot) [inline website/docs/d/lacp_member_policy.markdown:33]: of the LACP

**Review by lhercot — CHANGES_REQUESTED** [review]

**Comment by lhercot** [comment]
Also rebase with latest

File: website/docs/d/lacp_member_policy.markdown (hunk 40 above)
Comment (akinross) [reply website/docs/d/lacp_member_policy.markdown:33]: ok

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 41]
```diff
@@ -0,0 +1,279 @@
...
+
+	_, err = setOverridePCVPCPolicyGroupAttributes(infraAccBndlSubgrp, d)
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	infraRsLacpInterfacePolData := getInfraRsLacpInterfacePolData("Read", infraAccBndlSubgrp.DistinguishedName, aciClient)
+	if infraRsLacpInterfacePolData != nil {
+		d.Set("port_channel_member", fmt.Sprintf("uni/infra/lacpifp-%s", infraRsLacpInterfacePolData.(string)))
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:259]: ok

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 42]
```diff
@@ -0,0 +1,279 @@
...
+		return nil, err
+	}
+	schemaFilled, err := setOverridePCVPCPolicyGroupAttributes(infraAccBndlSubgrp, d)
+	if err != nil {
+		return nil, err
+	}
+
+	infraRsLacpInterfacePolData := getInfraRsLacpInterfacePolData("Import", infraAccBndlSubgrp.DistinguishedName, aciClient)
+	if infraRsLacpInterfacePolData != nil {
+		d.Set("port_channel_member", fmt.Sprintf("uni/infra/lacpifp-%s", infraRsLacpInterfacePolData.(string)))
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/resource_aci_infraaccbndlsubgrp.go:110]: ok

File: aci/data_source_aci_infraaccbndlsubgrp.go [hunk 43]
```diff
@@ -0,0 +1,59 @@
...
+	d.SetId(dn)
+
+	_, err = setOverridePCVPCPolicyGroupAttributes(infraAccBndlSubgrp, d)
+	if err != nil {
+		return nil
+	}
+
+	infraRsLacpInterfacePolData := getInfraRsLacpInterfacePolData("Read", infraAccBndlSubgrp.DistinguishedName, aciClient)
+	if infraRsLacpInterfacePolData != nil {
+		d.Set("port_channel_member", fmt.Sprintf("uni/infra/lacpifp-%s", infraRsLacpInterfacePolData.(string)))
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/data_source_aci_infraaccbndlsubgrp.go:55]: ok

File: aci/data_source_aci_infraaccbndlsubgrp.go [hunk 44]
```diff
@@ -0,0 +1,59 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"port_channel_member": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```
[the remark this answers is not in the document]
Comment (akinross) [reply aci/data_source_aci_infraaccbndlsubgrp.go:28]: ok

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**